### DISCUSSION TITLE
Is it possible to fille the "Outline View" from a extension

### DISCUSSION THREAD

**Discussion by jogibear9988** [microsoft/vscode-discussions#304] [Extension Development QnA] [answered]
I found this sample:

https://github.com/Microsoft/vscode-extension-samples/blob/main/tree-view-sample/USAGE.md

These show how to make your own Tree View, but I wanted to fill the existing Outline View.
Is this possible?

If not, how can I fill my own treeview, only when a specific document/view is active? (And fill it with data from this view)

**Comment by gjsjohnmurray** [comment]
I suggest you look at the DocumentSymbolProvider API.

**Reply by jogibear9988** [reply]
Will this work? I don't want to register a new language. 
I work on a visual HTML Designer, and want to show the DOM in the Outline view.

How can I tell VS to use a special documentSymbolProvider when my view is active?

**Comment by jogibear9988** [comment, marked as answer]
I think I go with a own treeview, so I also can use DragDrop: https://github.com/Microsoft/vscode/issues/32592
